lib/services.js

```diff
@@ -258,24 +260,30 @@ export default {
                     return cb(err);
                 }
                 omVal.acl = parsedACL;
-                metadata.putObjectMD(bucketName, objectKey, omVal, log, err => {
+                metadata.putObjectMD(bucketName, objectKey, omVal, log,
+                    (err, data) => {
+                        if (err) {
+                            log.error('error from metadata', { error: err });
+                            return cb(err);
+                        }
+                        log.trace('object successfully stored in metadata');
+                        const value = data ? JSON.parse(data) : {};
```

Comment:
Could we have a class for the data coming back with a deserialize method that is called in the wrapper?